src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -255,12 +255,21 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
             _workerChannelLogger.LogDebug("Received WorkerInitResponse. Worker process initialized");
             _initMessage = initEvent.Message.WorkerInitResponse;
             _workerChannelLogger.LogDebug($"Worker capabilities: {_initMessage.Capabilities}");
+
+            if (_initMessage.WorkerMetadata != null)
+            {
+                _initMessage.WorkerMetadata = ParseWorkerMetadata(_initMessage.WorkerMetadata);
```

Comment:
The object passed in gets updated, so it makes it a bit odd to reassign to this property. Perhaps updating the name (or adding the method to the message) would make this clearer

Reply:
Yeah I didn't really like this approach either - but adding the method to the message sounds like the nicest way to do this!